ebuild-writing/eapi/text.xml

```diff
@@ -616,6 +616,774 @@ installation targets.
+        </p>
+
+<pre>
+# old solution
+func() {
+	local prev_shopt=$(shopt -p -o noglob)
+	set -o noglob
+	${prev_shopt}
+}
+
+# new solution
+func() {
+	local -
+	set -o noglob
+}
+</pre>
+
+        <p>
+        The complete information on all changes and new features can be found
+        in the
+        <uri link="https://git.savannah.gnu.org/cgit/bash.git/tree/NEWS?h=bash-5.0">
+        bash 5.0 (and earlier) release notes</uri>.
+        </p>
+      </li>
+    </ul>
+  </dd>
+</dl>
+
+</body>
+</subsection>
+<subsection>
+<title>EAPI 8 variables</title>
+<body>
+
+<dl>
+  <dt>Selective fetch/mirror restriction</dt>
+  <dd>
+    <p>
+    Before EAPI 8, fetch and mirror restrictions applied globally. That is,
+    if you needed to apply the respective restriction to at least one distfile,
+    you had to apply it to them all. However, sometimes packages used a
+    combination of proprietary and free distfiles, the latter including e.g.
+    third party patches, artwork. They had to be mirror-restricted so far.
+    </p>
+
+    <p>
+    EAPI 8 brings the possibility of undoing fetch and mirror restriction
+    for individual files. The rough idea is that you put a <c>RESTRICT</c>
+    as before, then use the special <c>fetch+</c> prefix to specify URLs that
+    can be fetched from, or <c>mirror+</c> prefix to reenable mirroring of
+    individual files.
+    </p>
+
+    <p>
+    Similarly to how the <c>fetch</c> restriction implies a <c>mirror</c>
+    restriction, the <c>mirror</c> override implies a <c>fetch</c> override.
+    This might sound confusing at first but when you think about it, it's
+    perfectly logical.
+    </p>
+
+<codesample lang="ebuild">
+EAPI=8
+
+SRC_URI="
+	${P}.tgz
+	fetch+https://example.com/${P}-patch-1.tgz
+	mirror+https://example.com/${P}-fanstuff.tgz"
+
+RESTRICT="fetch"
+</codesample>
+
+    <p>
+    The following table summarizes it:
+    </p>
+
+    <table>
+      <tr>
+        <th><c>RESTRICT</c></th>
+        <th>URI prefix</th>
+        <th>Fetching</th>
+        <th>Mirroring</th>
+      </tr>
+      <tr>
+        <ti>(none)</ti>
+        <ti>(any)</ti>
+        <ti>allowed</ti>
+        <ti>allowed</ti>
+      </tr>
+      <tr>
+        <ti rowspan="2">mirror</ti>
+        <ti>(none) / fetch+</ti>
+        <ti>allowed</ti>
+        <ti>prohibited</ti>
+      </tr>
+      <tr>
+        <ti>mirror+</ti>
+        <ti>allowed</ti>
+        <ti>allowed</ti>
+      </tr>
+      <tr>
+        <ti rowspan="3">fetch</ti>
+        <ti>(none)</ti>
+        <ti>prohibited</ti>
+        <ti>prohibited</ti>
+      </tr>
+      <tr>
+        <ti>fetch+</ti>
+        <ti>allowed</ti>
+        <ti>prohibited</ti>
+      </tr>
+      <tr>
+        <ti>mirror+</ti>
+        <ti>allowed</ti>
+        <ti>allowed</ti>
+      </tr>
+    </table>
+  </dd>
+
+  <dt>Install-time dependencies (<c>IDEPEND</c>)</dt>
+  <dd>
+    <p>
+    Install-time dependencies are not a new idea. It has been on the board
```

Comment:
I think we should cut out the background for the most part, but make some reference to BDEPEND, of course.

I'd say some of this discussion about CBUILD should be shifted to a different section, *but* if you want to handle that another time, that's fine. It's already on my list.

Reply:
For now, I've removed the first paragraph and reworded the second one.